site/zenodo_rdm/moderation/handlers.py

```diff
@@ -185,7 +198,30 @@ def __init__(self):
 
     def publish(self, identity, draft=None, record=None, uow=None, **kwargs):
         """Calculate and log the score when a record is published."""
-        self.run(identity, record=record, uow=uow)
+        user_id = None
+        if identity == system_identity:
+            user_id = record.parent.access.owned_by.owner_id
+        else:
+            user_id = identity.id
+        if user_id is None:
+            current_app.logger.error(
+                "No user found for moderation action", stack_info=True
+            )
+            return
```

Comment:
Should this be return or raise?

I can't think of when are we expecting this.

Reply:
In general there's no "result" coming back form these calls, so the intent is to just "bail" early here.

Reply:
Also, not having a user is often when there's some complex workflow with nested service calls that somehow lose the `identity` on the way (which why above we use the record owner ID instead as a fallback)